src/potree.ts

```diff
@@ -167,13 +175,17 @@ export class Potree implements IPotree {
     } // end priority queue loop
 
     const numNodesToLoad = Math.min(this.maxNumNodesLoading, unloadedGeometry.length);
+    const promises: Promise<void>[] = [];
     for (let i = 0; i < numNodesToLoad; i++) {
-      unloadedGeometry[i].load();
+      promises.push(unloadedGeometry[i].load());
     }
 
     return {
       visibleNodes: visibleNodes,
       numVisiblePoints: numVisiblePoints,
+      exceededMaxLoadsToGPU: exceededMaxLoadsToGPU,
+      failed: failed,
+      promises: promises
```

Comment:
I don't think I like this name very much because it doesn't indicate what the promises are for. It would be better to have something like `loadPromises` or `nodeLoadPromisses`.

Reply:
True, `nodeLoadPromises` and `nodeLoadFailed`? (I'd rename the `failed` one, as well)

Reply:
Yep, makes sense.